Cache/IO.lean

```diff
@@ -406,4 +406,76 @@ def lookup (hashMap : ModuleHashMap) (paths : List FilePath) : IO Unit := do
       println! "  comment: {line}"
   if err then IO.Process.exit 1
 
+/--
+Parse command line arguments.
+Position `0` (i.e. the command `get`, `clean`, etc.) is ignored.
+The remaining arguments take one of the following forms:
+1. `Mathlib.Algebra.Fields.Basic`: there exists such a Lean file
+2. `Mathlib.Algebra.Fields`: no Lean file exists but a folder
+3. `Mathlib/Algebra/Fields/Basic.lean`: the file exists (note potentially `\` on Windows)
+4. `Mathlib/Algebra/Fields/`: the folder exists
+5. (`Aesop/Builder.lean`: the file does not exist, it's
+    actually somewhere in `.lake`. (not supported currently!))
+An argument like `Archive` is treated as module, not a path.
+-/
+def parseArgs (args : List String) : CacheM <| Std.HashMap Name FilePath := do
+  match args with
+  | [] => pure ∅
+  | _ :: args₀ =>
+    let sp := (← read).srcSearchPath
+    args₀.foldlM (init := ∅) fun acc (argₛ : String) => do
+      let arg : FilePath := argₛ
+      if arg.components.length > 1 || arg.extension == "lean" then
+        -- provided file name of a Lean file
+        let mod : Name := arg.withExtension "" |>.components.foldl .str .anonymous
+        let srcDir ← getSrcDir arg
+        if !(← arg.pathExists) then
+          IO.eprintln s!"Invalid argument: non-existing path {arg}"
+          IO.Process.exit 1
+        if arg.extension == "lean" then
+          -- provided existing `.lean` file
+          pure <| acc.insert mod argₛ
+        else
+          -- provided existing directory: walk it
+          IO.println s!"Searching directory {arg} for .lean files"
+          let leanModulesInFolder ← walkDir arg srcDir
+          pure <| acc.insertMany leanModulesInFolder
+      else
+        -- provided a module
+        let mod := argₛ.toName
+        let sourceFile ← Lean.findLean sp mod
+
+        if ← sourceFile.pathExists then
+          -- provided valid module
+          pure <| acc.insert mod sourceFile
+        else
+          -- provided "pseudo-module" like `Mathlib.Data` which
+          -- does not correspond to a Lean file, but to an existing folder
+          -- `Mathlib/Data/`
+          let folder := sourceFile.withExtension ""
+          IO.println s!"Searching directory {folder} for .lean files"
+          if ← folder.pathExists then
+            -- provided "module name" of an existing folder: walk dir
+            IO.eprintln <| "Entering a part of a module name " ++
+            "(i.e. `Mathlib.Data` when only the fold `Mathlib/Data/` but no " ++
+            "file `Mathlib/Data.lean` exists) is not supported yet!"
+            IO.Process.exit 1
+          else
+            IO.eprintln s!"Invalid argument: non-existing module {mod}"
+            IO.Process.exit 1
```

Comment:
Could you extract this as a helper function "leanFilesFromSpec` or similar, that parses just a single argument?